### PR TITLE
New auth0 credentials

### LINKED ISSUE #140: new Auth0 credentials
Quoting an email from Peter:

> Attention: mago, rehan, andy
> 
> jabba has now added http://localhost:3000 and https://delivery-console.dev.mozaws.net to the Auth0 config for Normandy Dev.
> 
> Please test now with client ID: hU1YpGcL82wL04vTPsaPAQmkilrSE7wr
> and domain auth.mozilla.auth0.com 
> 
> We've never done implicit grant (the opposite of code grant redirect dance) using this config before so it'll be an interesting test.
> 
> The Auth0 configuration requires a "Full URL" and for that jabba put in http://localhost:3000/ and https://delivery-console.dev.mozaws.net/
> Not sure if that matters or is relevant for implicit grant.
> 
> Also, I explained to jabba that we want to test Delivery Console Auth as soon as possible, but in the future we might have Delivery Console spanning not just Normandy but OtherBackend1 and AnotherBackend2, etc. When we get nearer that point, let's re-evaluate if we should mess more deeply with the Auth0 config or take a different approach.
>

### PR DESCRIPTION
**Pull request by magopian**
Fixes #140

### CODE REVIEW

File: src/console/App.js [hunk 1]
```diff
@@ -47,24 +47,33 @@ const CurrentUserInfo = props => {
   }
 };
 
+const Error = ({ error }) => {
```
Comment (andymikulski) [inline src/console/App.js]: IMO we should move away from using these 'functional components'. They're hard to find, add extra responsibilities to files, and this particular instance is not `export`ed, so there isn't any potential reuse later.

Comment (magopian) [reply src/console/App.js]: What would you use instead? The `export default class App extends Component` way?

Comment (andymikulski) [reply src/console/App.js]: Yep, make a new file for the `Error`, `export default class ErrorDisplay extends React.Component...`, then import it into `console/App.js`

Comment (peterbe) [reply src/console/App.js]: `extends React.PureComponent` ? :)

**Comment by magopian** [comment]
Rebased and fixed, what do you think @rehandalal ?

**Comment by magopian** [comment]
FYI there's a proptype error in the console when loging in, that I'm not sure where it's coming from:

screen shot 2018-04-30 at 11 41 16

**Comment by magopian** [comment]
That's an issue indeed. What should we do?
1/ ask Jabba if we can have a wildcard
2/ save the current page in the session/localstorage, then send the root url as the callback to auth0, then on the way back, redirect to the stored page

What do you think @rehandalal

**Comment by mostlygeek** [comment]
If the user goes to a page (`/recipe`) that requires they be authenticated they should be redirected back to `/`.  When they log in via auth0 it should redirect them back to `/`.

**Comment by peterbe** [comment]
> 1/ ask Jabba if we can have a wildcard

No. 

> 2/ save the current page in the session/localstorage

Yes please :)
Probably best to use sessionStorage so it doesn't surprise other tabs.

**Comment by magopian** [comment]
Ok, so `react-router` noob question: is there a way to get the absolute url of the **root** (the path to the homepage)?

I know we can access the current path with `this.props.location.pathname` from a component that is a children of a router component, but that would give me the **path** of the **curent** route (and not the path of the homepage).

If we're sure we're going to always deploy on `/`, I could simply use `window.location`...

Thoughts @rehandalal @peterbe ?

**Comment by magopian** [comment]
Following a discussion with @peterbe, it seems there's no way to get a url (or a path) for a route other than the current one in react-router.

So the current implementation is naive, but should work "well enough", and still allow us to change the `PUBLIC_URL` if needed when deploying in a subfolder for example.

What do you think @rehandalal @peterbe ?

**Comment by magopian** [comment]
Good call, I've also renamed `SET_USER_INFO` to `USER_SET_INFO`.

What do you think @rehandalal ?

**Comment by rehandalal** [comment]
nice work!

**Pull request merged**